static/src/javascripts/projects/commercial/modules/dfp/define-slot.js

```diff
@@ -0,0 +1,82 @@
+// @flow
+import { getUrlVars } from 'lib/url';
+import config from 'lib/config';
+import detect from 'lib/detect';
+import uniq from 'lodash/arrays/uniq';
+import flatten from 'lodash/arrays/flatten';
+import once from 'lodash/functions/once';
+import prepareSwitchTag from 'commercial/modules/dfp/prepare-switch-tag';
+
+const adUnit = once(() => {
+    const urlVars = getUrlVars();
+    return urlVars['ad-unit']
+        ? `/${config.page.dfpAccountId}/${urlVars['ad-unit']}`
+        : config.page.adUnit;
+});
+
+/**
+ * Builds and assigns the correct size map for a slot based on the breakpoints
+ * attached to the element via data attributes.
+ *
+ * A new size map is created for a given slot. We then loop through each breakpoint
+ * defined in the config, checking if that breakpoint has been set on the slot.
+ *
+ * If it has been defined, then we add that size to the size mapping.
+ *
+ */
+const buildSizeMapping = (sizes: Object) => {
+    const mapping = window.googletag.sizeMapping();
+
+    detect.breakpoints.filter(_ => _.name in sizes).forEach(_ => {
+        mapping.addSize([_.width, 0], sizes[_.name]);
+    });
+
+    return mapping.build();
+};
+
+const getSizeOpts = (sizesByBreakpoint: Object) => {
```

Comment:
There is quite a lot of horrible mangling of breakpoint-size mappings here. I've left it as-is apart from renaming some of the variables to better reflect the objects they point to (thereby avoiding a `shadow-variable: size` linting error.